example/c/boundary.c

```diff
@@ -0,0 +1,295 @@
+/* gcc boundary.c -o boundary -Wconversion -Wpointer-sign */
+#include <msgpack.h>
+#include <stdio.h>
+#include <assert.h>
+
+static inline unsigned char atohex(char a)
+{
+    int x;
+    if (a >= 'a') {
+        x = a - 'a' + 10;
+    } else if (a >= 'A') {
+        x = a - 'A' + 10;
+    } else {
+        x = a - '0';
+    }
+    assert(x >= 0 && x < 16);
+    return (unsigned char)x;
+}
+
+// Return 1 if equal
```

Comment:
It is not an essential issue but the function name `bytesncmp()` implies https://en.cppreference.com/w/c/string/byte/memcmp, so I think that it is better to the return value is the same as memcmp().